src/utils.h

```diff
@@ -72,6 +72,20 @@
     standard_ProcessUtility(PROCESS_UTILITY_ARGS);                             \
   }
 
+#define run_process_utility_hook_with_cleanup(process_utility_hook,            \
+                                              already_switched_to_superuser,   \
+                                              switch_to_original_role)         \
+  PG_TRY();                                                                    \
+  { run_process_utility_hook(prev_hook); }                                     \
+  PG_CATCH();                                                                  \
+  {                                                                            \
+    if (!already_switched_to_superuser) {                                      \
```

Comment:
It is recommended to use parenthesis if an argument is called with operators:
```suggestion
    if (!(already_switched_to_superuser)) {                                      \
```